torch/_dynamo/variables/torch_function.py

```diff
@@ -126,6 +126,30 @@ def subclass_type_var(self):
     def global_mangled_class_name(self):
         return f"__subclass_{self.class_type.__name__}_{id(self.class_type)}"
 
+    def var_getattr(self, tx, name):
+        # [Note: __torch_function__] We currently only support attributes that are defined on
+        # base tensors, custom attribute accesses will graph break. We will need to track setting
+        # attrs on this object.
+        import torch
+        from .builder import SourcelessBuilder
+
+        if name in banned_attrs:
+            unimplemented(
+                f"Accessing {name} on a tensor subclass with a __torch_function__ override is not supported"
+            )
+
+        if tx.output.torch_function_enabled:
+            get_fn = SourcelessBuilder()(tx, getattr(torch.Tensor, name).__get__)
```

Comment:
Check that `source` is not set?  Or are we sure no guards are needed?

Reply:
I can create a source for this since I'm grabbing an attribute from `torch.Tensor`

Reply:
I think we need some sort of check that the user defined class doesn't overload `name`.  If the tensor subclass defines `name`, then `getattr(torch.Tensor, name)` will be wrong.

Reply:
I checked if a property/method/attr is overridden on the class and trigger a graph break in this case. I added guards if the source is valid to ensure that the impl of a method/property/attr on the class matches the `torch.Tensor` impl as well. Handling Ed's case will be a bit trickier with `super()` handling for subclasses so I'm going to fallback for now.